### PR TITLE
 [DOCS]: Port over "How to instantiate a Data Context w/o YML" from RTD to Docusaurus

### PR DESCRIPTION
**Pull request by cdkini**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- Port over `.rst` guide to `.md`
-
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [ ] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [ ] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: docs/guides/setup/configuring-data-contexts/how-to-instantiate-a-data-context-without-a-yml-file.md [hunk 1]
```diff
@@ -1,5 +1,206 @@
...
+Steps
+-----
+
+1. **Create a DataContextConfig**
+
+    The DataContextConfig holds all of the associated configuration parameters to build a DataContext. There are defaults set for you to minimize configuration in typical cases, but please note that every parameter is configurable and all defaults are overridable. Also note that DatasourceConfig also has defaults which can be overridden.
+
+    Here we will show a few examples of common configurations, using the ``store_backend_defaults`` parameter. Note that you can continue with the existing API sans defaults by omitting this parameter, and you can override all of the parameters as shown in the last example. Note that a parameter set in ``DataContextConfig`` will override a parameter set in ``store_backend_defaults`` if both are used.
+
+    **TODO(cdkini): These are links to our API Reference, which has not yet been implemented. What should we do here?**
```
Comment (cdkini) [inline docs/guides/setup/configuring-data-contexts/how-to-instantiate-a-data-context-without-a-yml-file.md:28]: Please note this TODO here and below.

File: docs/guides/setup/configuring-data-contexts/how-to-instantiate-a-data-context-without-a-yml-file.md [hunk 2]
```diff
@@ -1,5 +1,206 @@
...
+
+    ```python
+    context = BaseDataContext(project_config=data_context_config)
+    ```
+
+3. **Use this BaseDataContext instance as your DataContext**
+
+    If you are using Airflow, you may wish to pass this Data Context to your GreatExpectationsOperator as a parameter. See the following guide for more details:
+
+    - [Deploying Great Expectations with Airflow](../../../guides/w) - **TODO(cdkini): Where do we link this?**
```
Comment (cdkini) [inline docs/guides/setup/configuring-data-contexts/how-to-instantiate-a-data-context-without-a-yml-file.md]: Please note this TODO here; we do not have this file in our new docs.

https://docs.greatexpectations.io/en/latest/guides/workflows_patterns/deployment_airflow.html#deployment-airflow

**Review by cdkini — COMMENTED** [review]
Screenshot 2021-07-09 at 14-06-39 How to instantiate a Data Context without a yml file — great_expectations documentation
Screenshot 2021-07-09 at 14-06-19 How to instantiate a Data Context without a yml file Great Expectations

**Review by alexsherstinsky — APPROVED** [review]
LGTM

**Pull request merged**